primitives/src/channel.rs

```diff
@@ -27,16 +27,21 @@ impl fmt::Debug for ChannelId {
     }
 }
 
-fn channel_id_from_str<'de, D>(deserializer: D) -> Result<[u8; 32], D::Error>
+fn deserialize_channel_id<'de, D>(deserializer: D) -> Result<[u8; 32], D::Error>
 where
     D: Deserializer<'de>,
 {
     let channel_id = String::deserialize(deserializer)?;
-    if channel_id.is_empty() || channel_id.len() != 66 {
-        return Err(serde::de::Error::custom("invalid channel id".to_string()));
-    }
+    let channel_id = validate_channel_id(channel_id).map_err(serde::de::Error::custom)?;
+    <[u8; 32] as FromHex>::from_hex(channel_id).map_err(serde::de::Error::custom)
+}
 
-    <[u8; 32] as FromHex>::from_hex(&channel_id[2..]).map_err(serde::de::Error::custom)
+fn validate_channel_id(s: String) -> Result<String, FromHexError> {
+    if s.is_empty() || s.len() != 66 || &s[0..2] != "0x" {
```

Comment:
Use `&str` instead of `String`, borrow is better, since you avoid unnecessary clones (expensive copies)

Be careful when dealing with indices like `[0..2]`, they usually `panic!` if something goes wrong.
Use Non-panicing methods instead:

https://doc.rust-lang.org/std/primitive.str.html#method.get

They also work with ranges `get(2..)` and will return `None` (`is_empty()`) instead of panicking.
Just use a `match (s.get(0..2), s.get(2..))`, you can also match on the prefix `(Some(prefix), Some(hex)) if hex.len() == 64`.

Also you can make the return a [u8; 32]
`<[u8; 32] as FromHex>::from_hex(hex)`